### ISSUE TITLE
sol?

### ISSUE THREAD

**Issue by ZhaoyuWang1** [JMH-codeking/OCNN#1]
def crossentropy2d(pred, target, weight=None, ignore_index=2, size_average=True):
    '''
    Parameters
    -----------
    pred : autograd.Variable. (N, C, H, W)
        where C is number of classes.
    target : (N, H, W), where all values 0 <= target[i] <= C-1.

    Returns
    -------
    loss : Tensor.
    '''
    # pred dims (N, C, H, W)
    n, c, h, w = pred.size()
    # log_p : log_probabilities (N, C, H, W)
    log_p = F.log_softmax(pred)
    # Linearize log_p
    # log_p : (N, C, H, W) --> (N*H*W, C)
    # move dim C over twice (N, C, H, W) > (N, H, C, W) > (N, H, W, C)
    log_p = log_p.transpose(1,2).transpose(2,3).contiguous()
    # (N, H, W, C) --> (N*H*W, C)
    log_p = log_p.view(-1, c)

    # Reshape target to a (N*H*W,), where each values 0 <= i <= C-1
    target = target.view(-1)
    loss = F.nll_loss(log_p, target, weight=weight,
                        size_average=True, ignore_index=ignore_index)

    return loss
